consensus/hotstuff/pacemaker.go

```diff
@@ -40,29 +40,28 @@ type PaceMaker interface {
 	// CurView returns the current view.
 	CurView() uint64
 
-	// UpdateCurViewWithQC will check if the given QC will allow PaceMaker to fast
-	// forward to QC.view+1. If PaceMaker incremented the current View, a NewViewEvent will be returned.
-	UpdateCurViewWithQC(qc *flow.QuorumCertificate) (*model.NewViewEvent, bool)
+	// ProcessQC will check if the given QC will allow PaceMaker to fast-forward to QC.view+1.
+	// If PaceMaker incremented the current View, a NewViewEvent will be returned.
+	ProcessQC(qc *flow.QuorumCertificate) (*model.NewViewEvent, bool)
 
-	// UpdateCurViewWithBlock will check if the given block will allow PaceMaker to fast forward
-	// to the BlockProposal's view. If yes, the PaceMaker will update it's internal value for
-	// CurView and return a NewViewEvent.
-	//
-	// The parameter `nextPrimary` indicates to the PaceMaker whether or not this replica is the
-	// primary for the NEXT view taking block.view as reference.
-	// True corresponds to this replica being the next primary.
-	UpdateCurViewWithBlock(block *model.Block, isLeaderForNextView bool) (*model.NewViewEvent, bool)
+	// ProcessTC will check if the given TC will allow PaceMaker to fast-forward to TC.view+1.
+	// If PaceMaker incremented the current View, a NewViewEvent will be returned.
+	ProcessTC(tc *flow.TimeoutCertificate) (*model.NewViewEvent, bool)
 
 	// TimeoutChannel returns the timeout channel for the CURRENTLY ACTIVE timeout.
-	// Each time the pace maker starts a new timeout, this channel is replaced.
+	// Each time the pacemaker starts a new timeout, this channel is replaced.
 	TimeoutChannel() <-chan time.Time
```

Comment:
I'm wondering if it still makes sense to have `TimeoutChannel` and `BlockRateDelay` methods in the `Pacemaker` interface. These methods call through directly to `TimeoutController`, and don't really need to involve `Pacemaker`. Should `EventHandler` use `TimeoutControl` directly instead?

Reply:
I would leave it as it is since we use only one function from it